include/aoapplication.h

```diff
@@ -126,16 +123,71 @@ class AOApplication : public QApplication
   QString get_evidence_path(QString p_file);
 
   /**
-   * @brief Searches for a file with any of the given extensions, and returns
-   * the first extension that actually matches to an existing file.
+   * @brief Searches for a file in any of the given paths, and returns
+   * the first path that actually matches to an existing file.
    *
-   * @param p_file The path to the file, without extension.
-   * @param p_exts The potential extensions the file could have.
+   * @param possible_paths Paths to check. Case-insensitive.
    *
-   * @return The first extension with which a file exists, or an empty string,
-   * if not one does.
+   * @return The first case-sensitive path that corresponds to an actual file,
+   * or an empty string, if not one does.
    */
-  QString get_file_extension(QString p_file, QVector<QString> p_exts);
+  QString find_asset_path(QStringList possible_paths);
+  /**
+   * @brief Searches for a file with any of the given root and extensions, and
+   * returns the first path that actually matches to an existing file. A root is
+   * matched to all given extensions in order before continuing to the next
+   * root.
+   *
+   * @param possible_roots The potential roots the filepath could have.
+   * Case-insensitive.
+   * @param possible_exts The potential extensions the filepath could have.
+   * Case-insensitive.
+   *
+   * @return The first case-sensitive  root+extension path for which a file
+   * exists, or an empty string, if not one does.
+   */
+  QString find_asset_path(QStringList possible_roots,
+                          QStringList possible_exts);
+
+  /**
+   * @brief Searches for a file in the current theme folder, and returns
+   * the first path that actually matches to an existing file.
+   * If `theme_folder = get_base_path() + "themes/"`
+   * the paths checked are (in order)
+   *
+   * 1. theme_folder + get_gamemode() + "/" + get_timeofday() + "/" + p_file
```

Comment:
This is implementation in the documentation. This may change in the future.
Instead, phrase these four lines in a way that does not hint at the implementation, like: "The time of day folder in the gamemode's folder in the theme's folder" or something more natural-sounding.

Reply:
I couldn't think of a nice way of doing a plain English description of the above, hence why I went with a C++ code description. I'll think of something